src/addrman_impl.h

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_ADDRMAN_IMPL_H
+#define BITCOIN_ADDRMAN_IMPL_H
+
+class AddrManImpl
+{
+public:
+    AddrManImpl(std::vector<bool>&& asmap, bool deterministic, int32_t consistency_check_ratio);
+
+    ~AddrManImpl();
+
+    template <typename Stream>
+    void Serialize(Stream& s_) const EXCLUSIVE_LOCKS_REQUIRED(!cs);
+
+    template <typename Stream>
+    void Unserialize(Stream& s_) EXCLUSIVE_LOCKS_REQUIRED(!cs);
+
+    size_t size() const EXCLUSIVE_LOCKS_REQUIRED(!cs);
+
+    bool Add(const std::vector<CAddress> &vAddr, const CNetAddr& source, int64_t nTimePenalty = 0)
```

Comment:
nit: I'd remove all the default arguments in AddrManImpl, at least in the public interface. Having defaults in both `AddrMan` and `AddrManImpl` can be a source of error when not both are updated

Reply:
definitely agree with this feedback! disagree with classifying it as a nit :)

I believe I removed all the defaults from the public interface, but I left them on the private functions because they aren't redundant in that situation.